File: lc49.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<int, IList<string>> map = new Dictionary<int, IList<string>>();
        foreach(string s in strs)
        {
            int code = GetModifiedString(s);
            //Console.WriteLine(code);
            if(!map.ContainsKey(code))
            {
                map.Add(code, new List<string>());
            }
            map[code].Add(s);
        }
        return map.Values.ToList();
    }
    private int GetModifiedString(string s)
    {
        int[] counts = new int[26];
        foreach(char c in s)
        {
            counts[c-'a']++;
        }
        return string.Join("#",counts).GetHashCode();
    }
}

File: python/lc49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = {}
        for word in strs:
            converted = str(sorted(word))
            if converted in m:
                m[converted].append(word)
            else:
                m[converted] = [word]
        # print(m.values())
        return list(m.values())
